projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.ts

```diff
@@ -62,6 +90,41 @@ export class CartItemComponent implements OnInit {
     );
   }
 
+  addToCart() {
```

Comment:
This component should by a __presentation component__. I would be careful with placing this feature logic in it. This component already getting complex with all the inputs/outputs.

I would say that this component should be refactored. It is used in a lot of places with very different goals. Making changes here confirms that this component is not extendable, thus should be looked into and rethought.